resolwe/flow/serializers/data.py

```diff
@@ -90,59 +89,61 @@ class Meta:
             'tags',
         )
 
-    def _serialize_items(self, serializer, kind, items):
-        """Return serialized items or list of ids, depending on `hydrate_XXX` query param."""
+    def _serialize_item(self, serializer, kind, item):
+        """Return serialized item or id, depending on `hydrate_XXX` query param."""
         if self.request and self.request.query_params.get('hydrate_{}'.format(kind), False):
-            serializer = serializer(items, many=True, read_only=True)
+            serializer = serializer(item, read_only=True)
             serializer.bind(kind, self)
             return serializer.data
         else:
-            return [item.id for item in items]
+            return item.id
 
-    def _filter_queryset(self, perms, queryset):
-        """Filter object objects by permissions of user in request."""
-        user = self.request.user if self.request else AnonymousUser()
-        return get_objects_for_user(user, perms, queryset)
+    @property
+    def user(self):
+        """Get user."""
+        return self.request.user if self.request else AnonymousUser()
 
-    def get_collection_names(self, data):
-        """Return serialized list of collection names on data that user has `view` permission on."""
-        collections = self._filter_queryset('view_collection', data.collection_set.all())
-        return list(collections.values_list('name', flat=True))
+    def get_collection_name(self, data):
```

Comment:
We can probably get rid of this now in favor of `entity` and `collection` fields on the data model.